File: 0621-task-scheduler/0621-task-scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        int totalJobs = tasks.size();
        int maxCount = 0;
        int X=0;
       unordered_map<char,int>map;
        for(char c:tasks){
            map[c]++;
            maxCount = max(maxCount,map[c]);
        }
        for(auto m:map){
            if(m.second==maxCount) X++;
        }
        return max(totalJobs, (maxCount-1)*(n+1)+X);
    }
};

File: 0621-task-scheduler/0621-task-scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        HashMap<Integer,Integer>map = new HashMap<>();
        int totalJob = tasks.length;
        int highestCount=0;
        for(int task:tasks){
            map.put(task, map.getOrDefault(task, 0)+1);
            highestCount=Math.max(highestCount,map.get(task));
        }
        int X=0;
        for(int val:map.values()){
            if(val==highestCount) X++;
        }

        return Math.max(totalJob, (highestCount-1)*(n+1)+X);

    }
}

